db/seeds.rb

```diff
@@ -243,77 +239,146 @@ def generate_blackout
+  res.equipment_item = res.equipment_model.equipment_items.all.sample
+  res.checkout_handler_id = User.where('role = ? OR role = ? OR role = ?',
+                                       'checkout', 'admin',
+                                       'superuser').all.sample.id
+end
+
+def make_returned(res, random = false)
+  # put reservation in the past
+  make_past(res, random)
+  # check out reservation
+  make_checked_out res
+
+  # check in reservation
+  res.status = 'returned'
+  len = res.due_date - res.start_date
+  offset = random ? rand(0...len).days : 0.days
+  res.checked_in = res.due_date - offset
+  res.checkin_handler_id = User.where('role = ? OR role = ? OR role = ?',
+                                      'checkout', 'admin',
+                                      'superuser').all.sample.id
+end
+
+def make_overdue(res, random = false)
+  # put reservation in the past
+  make_past(res, random)
+  # check out reservation
+  make_checked_out res
+
+  # ensure the reservation is actually overdue
+  # unlikely but necessary check due to randomness
+  if res.due_date >= Time.zone.today
+    len = res.due_date - res.start_date
+    res.due_date = Time.zone.today
+    res.due_date -= random ? rand(1.day..PAST_RANGE) : 1.day
+    res.start_date = res.due_date - len
+    travel_to(res.start_date) { res.save }
   end
+  res.overdue = true
+end
+
+def make_returned_overdue(res, random = false)
+  make_overdue(res, random)
+  len = Time.zone.today - res.due_date
+  offset = random ? rand(0...len).days : 0.days
+  res.status = 'returned'
+  res.checked_in = Time.zone.today - offset
+  res.checkin_handler_id = User.where('role = ? OR role = ? OR role = ?',
+                                      'checkout', 'admin',
+                                      'superuser').all.sample.id
+end
+
+def make_missed(res, random = false)
+  make_past(res, random)
+  res.status = 'missed'
+end
+
+def make_archived(res, random = false)
+  make_past(res, random) if random && rand < 0.5
+  res.status = 'archived'
+end
+
+def make_requested(res, random = false)
+  make_future(res, random) if random && rand < 0.5
+  res.status = 'requested'
+  res.flag(:request)
+end
+
+def make_denied(res, random = false)
+  make_requested(res, random)
+  res.status = 'denied'
+end
+
+def make_future(res, random = false)
+  # set the amount of time in the past
+  offset = random ? rand(1.day..FUTURE_RANGE) : 1.week
+  len = res.due_date - res.start_date
+  res.start_date += offset
+  res.due_date = res.start_date + len
 end
 
-def throw_into_past(res)
-  factor = rand(1.day..PAST_RANGE)
-  res.start_date = res.start_date - factor
-  res.due_date = res.due_date - factor
-  mark_checked_out res
-  mark_checked_in(res, res.equipment_model.category.max_checkout_length)
-  res.save(validate: false)
+def gen_res(random = false)
+  r = Reservation.new
+  r.status = 'reserved'
+  r.reserver_id = User.all.sample.id
+  r.equipment_model = EquipmentModel.all.sample
```

Comment:
Agreed, that makes sense - while it might slow us down a bit (unnecessary processing on invalid objects), I feel like architecturally having a single validation check makes more sense and a seed script that's marginally slower is acceptable given how infrequently it gets run, I think.